implementation/domain/logic/jbb-system-impl/src/test/java/org/jbb/system/impl/session/SessionServiceImplTest.java

```diff
@@ -0,0 +1,194 @@
+
+        //when
+        List<UserSession> allUserSessions = sessionService.getAllUserSessions();
+
+        //then
+        assertThat(allUserSessions.size()).isEqualTo(1);
+    }
+
+    @Test
+    public void whenTerminateSessionMethodIsInvokeThenUserSessionShouldBeClosed(){
+
+        //given
+        UserSession userSession = getSessionToDelete();
+        Map<String, ExpiringSession> fakeLogInUser = getLoggedUser();
+        Map<String, ExpiringSession> clearSessionMap = Maps.newHashMap();
+
+        when(jbbSessionRepository.getSessionMap()).thenReturn(fakeLogInUser);
+
+        //when
+        sessionService.terminateSession(userSession.sessionId());
+        when(jbbSessionRepository.getSessionMap()).thenReturn(clearSessionMap);
+        List<UserSession> allUserSessions = sessionService.getAllUserSessions();
+
+        //then
+        assertThat(allUserSessions.size()).isEqualTo(0);
+
+    }
+
+    @Test
+    public void whenSomeoneChangeDefaultInactiveSessionIntervalThenParameterShouldBeChangeWithoutException(){
+
+        //given
+        Duration newInterval = Duration.ofHours(1);
+        when(jbbSessionRepository.getDefaultMaxInactiveInterval()).thenReturn(3600);
+        //when
+        sessionService.setDefaultInactiveSessionInterval(newInterval);
+        Duration inactiveSessionInterval = sessionService.getDefaultInactiveSessionInterval();
+
+        //then
+        assertThat(inactiveSessionInterval.get(ChronoUnit.SECONDS)).isEqualTo(3600);
+    }
+
+    @Test
+    public void whenSomeoneSetNewValueOfPropertiesThenValueShouldBeSaveInFileAndInRepository(){
+
+        //given
+        int newValueOfProperties = 7200;
+
+        //when
+        sessionService.setDefaultInactiveSessionInterval(Duration.ofSeconds(newValueOfProperties));
+
+        //then
+        verify(systemProperties,atLeast(1)).setProperty(anyString(),anyString());
+        verify(jbbSessionRepository,atLeast(1)).setDefaultMaxInactiveInterval(anyInt());
+    }
+
+
+    private Map<String,ExpiringSession> getLoggedUser() {
+        Map<String,ExpiringSession> result = Maps.newHashMap();
+        MapSession mapSession = new MapSession();
+        String abc = "fakeuser";
+
+        SecurityContextImpl securityContext = Mockito.mock(SecurityContextImpl.class);
+
+        SecurityContentUser user = Mockito.mock(SecurityContentUser.class);
+        doReturn(abc).when(user).getDisplayedName();
+        doReturn(abc).when(user).getUsername();
+
+        Authentication authentication = Mockito.mock(Authentication.class);
+        when(authentication.getPrincipal()).thenReturn(user);
+        when(securityContext.getAuthentication()).thenReturn(authentication);
+
+        securityContext.setAuthentication(authentication);
+
+        mapSession.setAttribute("SPRING_SECURITY_CONTEXT",securityContext);
+        result.put("fakesessionid",mapSession);
+        return result;
+    }
+
+
+    private UserSession getSessionToDelete() {
+        return new UserSession() {
```

Comment:
Maybe extract this implementation of `UserSession` interface to test sources with name e.g. `TestbedUserSession`?